loxodrome_inverse in v2.7.1 fails depending on the geolocation order when arguments are 2d arrays and scalars

My following tests both passed in v2.7.0, but now the first one fails in v2.7.1.

```.py
import numpy as np
import pymap3d.lox as lox


def test_lox():
    lat_ref = 36.0
    lon_ref = 140.0
    lat = np.arange(30, 40).reshape(2, 5)
    lon = np.arange(135, 145).reshape(2, 5)
    len_, dir_ = lox.loxodrome_inverse(lat_ref, lon_ref, lat, lon)


def test_lox2():
    lat_ref = 36.0
    lon_ref = 140.0
    lat = np.arange(30, 40).reshape(2, 5)
    lon = np.arange(135, 145).reshape(2, 5)
    len_, dir_ = lox.loxodrome_inverse(lat, lon, lat_ref, lon_ref)
```

The failure is like this:

```
====================================================================== FAILURES =======================================================================
______________________________________________________________________ test_lox _______________________________________________________________________

    def test_lox():
        lat_ref = 36.0
        lon_ref = 140.0
        lat = np.arange(30, 40).reshape(2, 5)
        lon = np.arange(135, 145).reshape(2, 5)
>       len_, dir_ = lox.loxodrome_inverse(lat_ref, lon_ref, lat, lon)

tests/test_pymap3d.py:10: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../../../Library/Caches/pypoetry/virtualenvs/foo-TIPLNhpH-py3.9/lib/python3.9/site-packages/pymap3d/lox.py:173: in loxodrome_inverse
    lat2 = broadcast_to(lat2, lat1.shape)
<__array_function__ internals>:5: in broadcast_to
    ???
../../../../../Library/Caches/pypoetry/virtualenvs/foo-TIPLNhpH-py3.9/lib/python3.9/site-packages/numpy/lib/stride_tricks.py:411: in broadcast_to
    return _broadcast_to(array, shape, subok=subok, readonly=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

array = array([[0.52359878, 0.54105207, 0.55850536, 0.57595865, 0.59341195],
       [0.61086524, 0.62831853, 0.64577182, 0.66322512, 0.68067841]])
shape = (1,), subok = False, readonly = True

    def _broadcast_to(array, shape, subok, readonly):
        shape = tuple(shape) if np.iterable(shape) else (shape,)
        array = np.array(array, copy=False, subok=subok)
        if not shape and array.shape:
            raise ValueError('cannot broadcast a non-scalar to a scalar array')
        if any(size < 0 for size in shape):
            raise ValueError('all elements of broadcast shape must be non-'
                             'negative')
        extras = []
>       it = np.nditer(
            (array,), flags=['multi_index', 'refs_ok', 'zerosize_ok'] + extras,
            op_flags=['readonly'], itershape=shape, order='C')
E       ValueError: input operand has more dimensions than allowed by the axis remapping

../../../../../Library/Caches/pypoetry/virtualenvs/foo-TIPLNhpH-py3.9/lib/python3.9/site-packages/numpy/lib/stride_tricks.py:348: ValueError

```

I have a fix, so I will post a PR.

--- a/src/pymap3d/lox.py
+++ b/src/pymap3d/lox.py
@@ -13,6 +13,7 @@ try:
         pi,
         array,
         atleast_1d,
+        broadcast_arrays,
         broadcast_to,
     )
 except ImportError:
@@ -150,28 +151,7 @@ def loxodrome_inverse(
         lat1, lon1, lat2, lon2 = radians(lat1), radians(lon1), radians(lat2), radians(lon2)
 
     try:
-        lat1 = atleast_1d(lat1)
-        lon1 = atleast_1d(lon1)
-        lat2 = atleast_1d(lat2)
-        lon2 = atleast_1d(lon2)
-
-        if lat1.shape != lon1.shape:
-            if lat1.size == 1:
-                lat1 = broadcast_to(lat1, lon1.shape)
-            elif lon1.size == 1:
-                lon1 = broadcast_to(lon1, lat1.shape)
-            else:
-                raise ValueError("lat1, lon1 must have same shape")
-        if lat2.shape != lon2.shape:
-            if lat2.size == 1:
-                lat2 = broadcast_to(lat2, lon2.shape)
-            elif lon2.size == 1:
-                lon2 = broadcast_to(lon2, lat2.shape)
-            else:
-                raise ValueError("lat2, lon2 must have same shape")
-        if lat2.shape != lat1.shape:
-            lat2 = broadcast_to(lat2, lat1.shape)
-            lon2 = broadcast_to(lon2, lat1.shape)
+        lat1, lon1, lat2, lon2 = broadcast_arrays(lat1, lon1, lat2, lon2)
 
     except NameError:
         pass

--- a/src/pymap3d/tests/test_rhumb.py
+++ b/src/pymap3d/tests/test_rhumb.py
@@ -76,6 +76,25 @@ def test_numpy_loxodrome_inverse():
     d, a = lox.loxodrome_inverse([40, 40], [-80, -80], 65, [-148, -148])
 
 
+def test_numpy_2d_loxodrome_inverse():
+    pytest.importorskip("numpy")
+    d, a = lox.loxodrome_inverse(
+        [[40, 40], [40, 40]], [[-80, -80], [-80, -80]], 65, -148
+    )
+    assert d == approx(5248666.209)
+    assert a == approx(302.00567)
+
+    d, a = lox.loxodrome_inverse(
+        [[40, 40], [40, 40]], [[-80, -80], [-80, -80]], [[65, 65], [65, 65]], -148
+    )
+    d, a = lox.loxodrome_inverse(
+        [[40, 40], [40, 40]], [[-80, -80], [-80, -80]], 65, [[-148, -148], [-148, -148]]
+    )
+    d, a = lox.loxodrome_inverse(
+        40, -80, [[65, 65], [65, 65]], [[-148, -148], [-148, -148]]
+    )
+
+
 @pytest.mark.parametrize(
     "lat0,lon0,rng,az,lat1,lon1",
     [

